src/cobalt.rs

```diff
@@ -20,6 +24,45 @@ use config::SassOutputStyle;
 use files::FilesBuilder;
 use frontmatter;
 
+fn deep_insert(data_map: &mut HashMap<String, Value>,
+               file_path: &Path,
+               target_key: String,
+               data: Value)
+               -> Result<()> {
+    // now find the nested map it is supposed to be in
+    let target_map = if let Some(path) = file_path.parent() {
+        let mut map = data_map;
+        for part in path.iter() {
+            if part.len() == 0 {
+                continue
+            }
+            if let Some(key) = part.to_str() {
+                let cur_map = map;
+                map = cur_map
+                    .entry(String::from(key))
+                    .or_insert_with(|| Value::Object(HashMap::new()))
+                    .as_object_mut()
+                    .unwrap();
```

Comment:
Note: this is the main item of concern.

We should probably error rather than unwrap in case there are overlapping data files with different data types.

`_data/data.json`:
```js
{ "key": "value" }
```
`_data/data/key.json`:
```js
{ "a": "b" }
```